pkg/sdn/plugin/pod_linux.go

```diff
@@ -254,20 +191,88 @@ func (m *podManager) ipamDel(id string) error {
 	return nil
 }
 
-func isScriptError(err error) bool {
-	_, ok := err.(*exec.ExitError)
-	return ok
+func ensureOvsPort(ovsif *ovs.Interface, hostVeth string) (int, error) {
+	return ovsif.AddPort(hostVeth, -1)
 }
 
-// Get the last command (which is prefixed with "+" because of "set -x") and its output
-func getScriptError(output []byte) string {
-	lines := strings.Split(string(output), "\n")
-	for n := len(lines) - 1; n >= 0; n-- {
-		if strings.HasPrefix(lines[n], "+") {
-			return strings.Join(lines[n:], "\n")
+func setupPodFlows(ovsif *ovs.Interface, ofport int, podIP, podMac string, vnid uint32) error {
+	otx := ovsif.NewTransaction()
+
+	// ARP/IP traffic from container
+	otx.AddFlow("table=2, priority=100, in_port=%d, arp, nw_src=%s, arp_sha=%s, actions=load:%d->NXM_NX_REG0[], goto_table:5", ofport, podIP, podMac, vnid)
+	otx.AddFlow("table=2, priority=100, in_port=%d, ip, nw_src=%s, actions=load:%d->NXM_NX_REG0[], goto_table:3", ofport, podIP, vnid)
+
+	// ARP request/response to container (not isolated)
+	otx.AddFlow("table=6, priority=100, arp, nw_dst=%s, actions=output:%d", podIP, ofport)
+
+	// IP traffic to container
+	if vnid == 0 {
+		otx.AddFlow("table=7, priority=100, ip, nw_dst=%s, actions=output:%d", podIP, ofport)
+	} else {
+		otx.AddFlow("table=7, priority=100, reg0=0, ip, nw_dst=%s, actions=output:%d", podIP, ofport)
+		otx.AddFlow("table=7, priority=100, reg0=%d, ip, nw_dst=%s, actions=output:%d", vnid, podIP, ofport)
+	}
+
+	return otx.EndTransaction()
+}
+
+func setupPodBandwidth(ovsif *ovs.Interface, pod *kapi.Pod, hostVeth string) error {
```

Comment:
Yeah, looks good.  I find a lot of the time it's easy to "over OO" stuff, and I think for many functions you end up with more testable and less coupled code.